jolt-core/src/subprotocols/sumcheck.rs

```diff
@@ -520,19 +520,14 @@ impl<F: JoltField> SumcheckInstanceProof<F> {
         // verify that there is a univariate polynomial for each round
         assert_eq!(self.compressed_polys.len(), num_rounds);
         for i in 0..self.compressed_polys.len() {
-            let poly = self.compressed_polys[i].decompress(&e);
-
             // verify degree bound
-            if poly.degree() != degree_bound {
+            if self.compressed_polys[i].degree() != degree_bound {
                 return Err(ProofVerifyError::InvalidInputLength(
                     degree_bound,
-                    poly.degree(),
+                    self.compressed_polys[i].degree(),
                 ));
             }
 
-            // check if G_k(0) + G_k(1) = e
-            assert_eq!(poly.eval_at_zero() + poly.eval_at_one(), e);
-
```

Comment:
we still need this check no? The verifier needs to check that the current round's polynomial is consistent with the claim from the previous round